recipes/redis-plus-plus/all/conanfile.py

```diff
@@ -0,0 +1,80 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class RedisPlusPlusConan(ConanFile):
+    name = "redis-plus-plus"
+    homepage = "https://github.com/sewenew/redis-plus-plus"
+    description = "Redis client written in C++"
+    topics = ("conan", "database", "redis", "client", "tls")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False], "with_tls": [True, False], "with_cxx": [11, 17]}
+    default_options = {"shared": False, "fPIC": True, "with_tls": False, "with_cxx": 11}
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+            if self.settings.compiler == "Visual Studio":
+                raise ConanInvalidConfiguration("redis-plus-plus does not support begin compiled as shared on Visual Studio")
+
+    def validate(self):
+        if self.options.with_tls != self.options["hiredis"].with_ssl:
+            raise ConanInvalidConfiguration("with_tls must match hiredis.with_ssl option")
+
+    def requirements(self):
+        self.requires("hiredis/1.0.0")
+        if self.options.with_tls:
+            self.requires("openssl/1.1.1i")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["REDIS_PLUS_PLUS_CXX_STANDARD"] = self.options.with_cxx
```

Comment:
C++ standard can be overridden by settings.compiler.cppstd value. I wouldn't consider this options, instead, any use can set cppstd.

Reply:
Hmm, I'll to test that locally since I've never tried it